pre-emails/get_info.py

```
@@ -0,0 +1,322 @@
+# ── Image fetch ───────────────────────────────────────────────────────────────
+
+def fetch_image(url: str) -> str:
+    """Download image to pre-emails/images/ and return the local path."""
+    images_dir = Path(__file__).resolve().parent / "images"
+    url_path = url.split("?")[0].rstrip("/")
+    ext = Path(url_path).suffix or ".jpg"
+    filename = Path(url_path).name or f"cover{ext}"
+    local_path = images_dir / filename
+
+    try:
+        req = urllib.request.Request(url, headers={"User-Agent": "Mozilla/5.0"})
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)